yarn-project/foundation/src/abi/encoder.ts

```diff
@@ -33,8 +33,6 @@ class ArgumentEncoder {
             this.flattened.push(Fr.fromBuffer(arg));
           } else if (typeof arg.toField === 'function') {
             this.flattened.push(arg.toField());
-          } else if (arg instanceof Fr) {
-            this.flattened.push(arg);
```

Comment:
Fun fact: we were failing to encode fields as fields because the instanceof check sometimes failed due to more than one version of the class being loaded. Removed this in favor of having `toField` in the Fr and Fq classes.